db/compaction/compaction_iterator.cc

```diff
@@ -1101,7 +1118,11 @@ void CompactionIterator::DecideOutputLevel() {
                            &context);
   output_to_penultimate_level_ = context.output_to_penultimate_level;
 #endif /* !NDEBUG */
-
+  if (is_range_del_) {
+    // range tombstones are in penultimate level only
+    output_to_penultimate_level_ = true;
+    return;
+  }
```

Comment:
Good catch, I realize we should not emit/consider range tombstones for file cutting if they are to be dropped. This can happen in non-tiering settings too.

Reply:
Removed, and updated CompactionOutput to only consider range tombstone keys for file cutting when compaction output is not to bottomostlevel, otherwise there is no grandparent to overlap (or that grandparent will be changed in the case of tiered compaction).